src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,276 @@
+import { AfterContentChecked, Directive, ElementRef, Input, OnChanges, OnDestroy, Renderer2, SimpleChanges } from '@angular/core';
+import { ColorService } from './../../services/color/color.service';
+import { ThemeColor } from './../../services/color/theme-color';
+import { ContrastService } from './../accessibility/contrast-ratio/contrast.service';
+
+export type BadgeVerticalPosition = 'above' | 'below' | 'top' | 'bottom';
+
+export type BadgeHorizontalPosition = 'before' | 'after' | 'left' | 'right';
+
+export type BadgeSize = 'small' | 'medium' | 'large';
+
+@Directive({
+    selector: '[uxBadge]',
+    exportAs: 'ux-badge',
+    host: {
+        class: 'ux-badge-container',
+        '[class.ux-badge-small]': '_badgeSize === "small"',
+        '[class.ux-badge-medium]': '_badgeSize === "medium"',
+        '[class.ux-badge-large]': '_badgeSize === "large"',
+        '[class.ux-badge-above]': 'isAbove()',
```

Comment:
We can only reference public properties and functions in an Angular production build, otherwise it will throw errors.

Ideally we want to also just use equality checks here rather than function calls. Function calls are really slow in templates (and host bindings) and they get called every time change detection runs. Its often not a noticeable different in our docs site, however some of the apps our components are used in are really heavy enterprise pages that are not well optimized, so making our code efficient can make a big difference overall. Something like `'[class.ux-badge-above]': 'badgeVerticalPosition === "above" || badgeVerticalPosition === "top"` is much more efficient.